### PR TITLE
Make more String tests run on Linux

### PR DESCRIPTION
**Pull request by glessard**
This PR moves a number of String-related tests from `test/1_stdlib/NSStringAPI.swift`, where they couldn't run because the file requires `objc_interop`.

A few test cases have new xfail annotations, related to sort order (https://bugs.swift.org/browse/SR-530) and comparisons involving embedded nulls (https://bugs.swift.org/browse/SR-630).

### CODE REVIEW

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by glessard** [comment]
I don't know what made the CI tests fail, but I had left a `.skip` behind that shouldn't have been there. Now removed.

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by glessard** [comment]
The non-Linux failures don't seem directly related to this PR; should another test be run?

**Comment by gribozavr** [comment]
@swift-ci Please test

File: test/1_stdlib/StringAPI.swift [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+    expected.isEQ(), lhs, rhs, stackTrace: stackTrace.withCurrentLoc())
+
+  expectEqual(expected.isLT(), lhs < rhs, stackTrace: stackTrace)
+  expectEqual(expected.isLE(), lhs <= rhs, stackTrace: stackTrace)
+  expectEqual(expected.isGE(), lhs >= rhs, stackTrace: stackTrace)
+  expectEqual(expected.isGT(), lhs > rhs, stackTrace: stackTrace)
+  checkComparable(expected, lhs, rhs, stackTrace: stackTrace.withCurrentLoc())
+}
+
+for test in comparisonTests {
```
Comment (gribozavr) [inline test/1_stdlib/StringAPI.swift]: Why did you move the loop out of the test?  Doesn't the test provide the reference to the test data when it fails?

Comment (gribozavr) [inline test/1_stdlib/StringAPI.swift]: Oh, I see -- you added custom XFAILs.  Never mind.

Comment (glessard) [inline test/1_stdlib/StringAPI.swift]: I thought this would be better than duplicating the tests just for xfail cases.
Thanks for reviewing.

**Comment by gribozavr** [comment]
@glessard Thank you for this!

**Pull request merged**